test/test_linalg.py

```diff
@@ -3700,6 +3700,9 @@ def test_matrix_rank_atol_rtol(self, device, dtype):
             result = torch.linalg.matrix_rank(a, atol=tol_value, rtol=tol_value)
             self.assertEqual(result, 2)  # there are 2 singular values above max(0.81, 1.5*0.81)
 
+    # CUDA 11.6 issue failure https://github.com/pytorch/pytorch/issues/75391
+    @skipCUDAIf(torch.version.cuda is not None
+                and torch.version.cuda.split(".") == ["11", "6"], "There's a bug in CUDA 11.6")
```

Comment:
Thank you for submitting the issue.
This test verifies the behavior for tensors with 0 number of elements and the failure comes from the `matrix_rank(input, hermitian=True)` call.